src/item_box.py

```diff
@@ -98,5 +99,9 @@ def delete(self, _button):
         )
 
     @Gtk.Template.Callback()
-    def edit(self, _):
+    def edit(self, _button):
         EditItemWindow(self.props.application, self.props.item)
+
+    @Gtk.Template.Callback()
+    def on_label_change(self, _itembox, _gobject):
+        self.label.set_text(utilities.shorten_label(self.item.name))
```

Comment:
I'm not sure this logic is the way to go.

First, when an item's name is updated, the label instantly gets updated to the same value (potentially stretching the sidebar in the process).
Then the signal listener queries the items property and sets a shortened version, reducing the width again.

Using an intermediary function (e. g. like in #420), this issue can be avoided, by shortening the label whilst it gets copied from the item.